### PR TITLE
test(developer): kmcmplib compiler unit tests 5

### PR DESCRIPTION
**Pull request by markcsinclair**
Google Test based unit tests for Compiler.cpp

As it was identified that no recursive call was made to `GetXStringImpl()` from within the virtual key case (case 11), the checks (`Compiler.cpp`) and resulting error messages (`kmn_compiler_errors.h`) when other keywords were identified within a virtual key section were stripped out. The error messages removed were:
- `ERROR_StringInVirtualKeySection`
- `ERROR_AnyInVirtualKeySection`
- `ERROR_BeepInVirtualKeySection`
- `ERROR_IndexInVirtualKeySection`
- `ERROR_OutsInVirtualKeySection`
- `ERROR_ContextInVirtualKeySection`
- `ERROR_CallInVirtualKeySection`
- `ERROR_InvalidInVirtualKeySection`

In addition, three variables relating to the parsing of virtual keys were localised by turning the case into a block.

The corresponding errors were also removed from `kmn_compiler_messages.ts`.

TODO comments in `gtest-compiler-test.cpp` were also removed.

A separate issue and fix were not written, but the compiler changes incorporated here.

Continues PR #12489 

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by markcsinclair** [comment]
Ready to review

File: developer/src/kmcmplib/src/Compiler.cpp [hunk 1]
```diff
@@ -2413,166 +2399,172 @@ KMX_DWORD GetXStringImpl(PKMX_WCHAR tstr, PFILE_KEYBOARD fk, PKMX_WCHAR str, KMX
...
-      //printf("--EXTENDEDEND--\n");
+        tstr[mx++] = UC_SENTINEL_EXTENDEDEND;
+        tstr[mx] = 0;
+        //printf("--EXTENDEDEND--\n");
 
-      p = q + 1;
+        p = q + 1;
 
-      sFlag = 0;
+        sFlag = 0;
```
Comment (mcdurdin) [inline developer/src/kmcmplib/src/Compiler.cpp:2566]:
```suggestion
```

nit only :grin:

**Review by mcdurdin — APPROVED** [review]
LGTM, nice little cleanup

File: developer/src/kmcmplib/src/Compiler.cpp (hunk 1 above)
Comment (markcsinclair) [reply developer/src/kmcmplib/src/Compiler.cpp:2566]: The more unneeded code removed the better.

Comment (SabineSIL) [reply developer/src/kmcmplib/src/Compiler.cpp]: LGTM also

**Comment by mcdurdin** [comment]
> P;ease note that the code for `case 11` in `Compiler.cpp` is not changed as much as it appears - github has failed to spot the main change is just indentation by one level as it has been converted into a block.

You can use 'hide whitespace' in the configuration cog to make it easier to review:
image

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.146-alpha](https://keyman.com/downloads/releases/alpha/18.0.146)

**Pull request merged**